kubectl-plugin/pkg/util/generation/generation.go

```diff
@@ -26,40 +36,67 @@ type RayClusterSpecObject struct {
 	RayVersion *string `yaml:"ray-version,omitempty"`
 	Image      *string `yaml:"image,omitempty"`
 
-	HeadCPU              *string           `yaml:"head-cpu,omitempty"`
-	HeadGPU              *string           `yaml:"head-gpu,omitempty"`
-	HeadMemory           *string           `yaml:"head-memory,omitempty"`
-	HeadEphemeralStorage *string           `yaml:"head-ephemeral-storage,omitempty"`
-	HeadRayStartParams   map[string]string `yaml:"head-ray-start-params,omitempty"`
-	HeadNodeSelectors    map[string]string `yaml:"head-node-selectors,omitempty"`
+	Head *Head `yaml:"head,omitempty"`
```

Comment:
nested head attributes into `Head` struct and removed `head-` prefix from the YAML keys.